include/rtservice.h

```diff
@@ -254,6 +272,14 @@ rt_inline int rt_slist_isempty(rt_slist_t *l)
 #define rt_slist_entry(node, type, member) \
     rt_container_of(node, type, member)
 
+/**
+ * rt_slist_for_each  -   iterate over a single list
+ * @pos:    the rt_slist_t * to use as a loop cursor.
+ * @head:   the head for your single list.
+ */
+#define rt_slist_for_each(pos, head) \
+    for (pos = (head)->next; &pos->next != (head); pos = pos->next)
```

Comment:
for a single list, the for each may be is `for (pos = (head)->next; pos != (RT_NULL); pos = pos->next)`